verified_boot.md

```diff
@@ -0,0 +1,240 @@
+- Most updates also need to update the bootblock, so they require protection
+  to be disabled for updates - obviously suboptimal
+- OS secure boot is handled by UEFI SB
+
+???
+
+- Firmware programs protected range registers in chipset, which ensures software
+  cannot write to flash
+- Firmware is not physically write protected
+- We find we need to update the recovery region because of:
+  - Firmware layout changes
+  - New features added to booblock
+  - Breaking Kconfig changes
+  - coreboot rebases
+- Google is not affected by this because they:
+  - Generally do not add new features to firmware
+  - Use a fixed coreboot rev for each mainboard
+
+---
+
+# Problem
+
+- How do we improve Dasharo verified boot while staying secure and without
+  taking control away from users?
+  - Security: A guarantee that only firmware from a trusted vendor is run
+  - Control: Ability to self-sign, to inspect and replace firmware components
+- There are other verified boot schemes
+
+---
+
+# Intel Boot Guard
+
+- The most widely known option
+- Verifies and measures the initial bootblock
+- Ensures FW authenticity using keys fused to the chipset
+- Supports firmware verification and measurement
+- Different profiles with different features enabled
+  - Verified, Measured and Enforced knobs
+  - Verified boot is always enabled in all profiles
+- Ties a platform to a specific firmware vendor, forever
+
+???
+
+- Intel specific, AMD has a different but comparable secure boot scheme
+  Verifies the initial bootblock, rest is up to BIOS (Vboot)
+- fused: programmed into one-time programmable e-fuses inside the chipset
+- Measurements are made in locality 3, so they can't be faked by malicious BIOS
+- Profile 3: Verified and Measured with infinite time for remediation
+- Profile 3 acts as root of trust for measurement, but not for verification
+- On verification failure, the PCR0 measurements are always the same. This was
+  determined experimetally.
+- so users can choose to use Dasharo firmware and get measured IBB, or flash
+  their own and lose measured IBB.
+  - might make sense for some small percentage for users
+  - does not help us a lot
+
+---
+
+# Boot Guard cons
+
+- Requires more blobs in firmware
+  - Boot Guard ACM
+- Fusing removes some owner control
+- Profile 3 does not ensure initial boot block authenticity
+  - Only helps us establish a root of trust for measurement
+- Other profiles take away owner control completely
+
+???
+
+- ACM - Authenticated Code Module. Hyper-privileged, Intel signed blob that
+  performs the actual BIOS verification and measurement, then hands off to the
+  BIOS reset vector.
+
+---
+
+# CBFS Verification
+
+- A relatively new coreboot feature
+- Cryptographically verifies components of the coreboot image
+- According to documentation:
+  > This only makes sense if you use some
+  > out-of-band mechanism to guarantee the integrity of the bootblock
+  > itself, such as Intel Boot Guard or flash write-protection.
```

Comment:
Probably worth mentioning that this is the same as with Vboot.

Reply:
https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR175